### PR TITLE
Feat/esm tree shaking

### PR DESCRIPTION
**Pull request by corporateanon**
### Added
- #2 tree-shakeable ESM build under `./dist/esm`
- #2 integration test for the Webpack bundle with tree-shaking
- #2 integration test for the UMD bundle
- #2 integration test for the CommonJS module

### Change
- #25, #2 re-organize dist folder. `node` renamed to `cjs`, `browser` is removed, `esm` is added

To enable tree-shaking in **Webpack**, the following options should be added to the Webpack configuration:

```
    optimization: {
        innerGraph: true,
        usedExports: true,
    },

```

Then, if you try to build the following code with Webpack:

```js
import { translate, MOSMETRO, WIKIPEDIA } from 'iuliia';
console.log(translate("Привет, мир!", WIKIPEDIA));
console.log(translate("Привет, мир!", MOSMETRO)); 
```

, then your bundle will contain only `MOSMETRO` and `WIKIPEDIA` schemas because you enabled dead-code elimination optimization.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -4,30 +4,58 @@ import { Schemas } from "./schemas";
...
+export const MOSMETRO = /*#__PURE__*/ Schema.load(mosmetro);
+export const MVD_310 = /*#__PURE__*/ Schema.load(mvd_310);
+export const MVD_310_FR = /*#__PURE__*/ Schema.load(mvd_310_fr);
+export const MVD_782 = /*#__PURE__*/ Schema.load(mvd_782);
+export const SCIENTIFIC = /*#__PURE__*/ Schema.load(scientific);
+export const TELEGRAM = /*#__PURE__*/ Schema.load(telegram);
+export const UNGEGN_1987 = /*#__PURE__*/ Schema.load(ungegn_1987);
+export const WIKIPEDIA = /*#__PURE__*/ Schema.load(wikipedia);
+export const YANDEX_MAPS = /*#__PURE__*/ Schema.load(yandex_maps);
+export const YANDEX_MONEY = /*#__PURE__*/ Schema.load(yandex_money);
```
Comment (corporateanon) [inline src/index.ts:61]: `#__PURE__` annotation informs Terser that an expression does not produce side effects. It allows to drop the code if it is not used in the final application bundle.

File: package.json [hunk 2]
```diff
@@ -42,18 +42,20 @@
     "files": [
         "dist/**/*"
     ],
+    "sideEffects": false,
```
Comment (corporateanon) [inline package.json]: a field that informs Webpack that this package has no files which produce side effects when imported

**Review by nalgeon — APPROVED** [review]
This is somewhat beyond my JS capabilities, but I like what I see ツ

File: package.json [hunk 3]
```diff
@@ -16,6 +10,36 @@
...
     ],
+    "repository": {
+        "type": "git",
+        "url": "git+https://github.com/nalgeon/iuliia-js.git"
+    },
+    "license": "MIT",
+    "author": "Anton Zhiyanov <m@antonz.ru>",
+    "sideEffects": false,
+    "main": "dist/cjs/index.js",
+    "module": "dist/esm/index.js",
```
Comment (corporateanon) [inline package.json:21]: module is for client bundlers supporting ES modules (for example Webpack)

**Pull request merged**